internal/web/server.go

```diff
@@ -638,59 +807,68 @@ func (s *server) advancedConfiguration(w http.ResponseWriter, r *http.Request) {
-		if err := s.repoClientset.ForRepoWithName(repositoryName).FetchPackageIndex(pkgName, &idx); err != nil {
+		if err := s.repoClientset.ForRepoWithName(d.repositoryName).FetchPackageIndex(d.manifestName, &idx); err != nil {
 			s.respondAlertAndLog(w, err,
-				fmt.Sprintf("An error occurred fetching package index of %v in repository %v", pkgName, repositoryName),
+				fmt.Sprintf("An error occurred fetching package index of %v in repository %v", d.manifestName, d.repositoryName),
 				"danger")
 			return
 		}
 		latestVersion := idx.LatestVersion
 
-		if selectedVersion == "" {
-			selectedVersion = latestVersion
+		if d.selectedVersion == "" {
+			d.selectedVersion = latestVersion
 		} else if !slices.ContainsFunc(idx.Versions, func(item types.PackageIndexItem) bool {
-			return item.Version == selectedVersion
+			return item.Version == d.selectedVersion
 		}) {
-			selectedVersion = latestVersion
+			d.selectedVersion = latestVersion
 		}
 
-		res, err := s.dependencyMgr.Validate(r.Context(), manifest, selectedVersion)
+		res, err := s.dependencyMgr.Validate(r.Context(), d.manifest, d.selectedVersion)
 		if err != nil {
 			s.respondAlertAndLog(w, err,
-				fmt.Sprintf("An error occurred validating dependencies of %v in version %v", pkgName, selectedVersion),
+				fmt.Sprintf("An error occurred validating dependencies of %v in version %v", d.manifestName, d.selectedVersion),
 				"danger")
 			return
 		}
 
 		err = s.templates.pkgConfigAdvancedTmpl.Execute(w, s.enrichPage(r, map[string]any{
-			"Status":           client.GetStatusOrPending(pkg),
-			"Manifest":         manifest,
+			"Status":           client.GetStatusOrPending(d.pkg),
+			"Manifest":         d.manifest,
 			"LatestVersion":    latestVersion,
 			"ValidationResult": res,
 			"ShowConflicts":    res.Status == dependency.ValidationResultStatusConflict,
-			"SelectedVersion":  selectedVersion,
+			"SelectedVersion":  d.selectedVersion,
 			"PackageIndex":     &idx,
 			"Repositories":     repos,
-			"RepositoryName":   repositoryName,
+			"RepositoryName":   d.repositoryName,
+			"SelfHref":         fmt.Sprintf("%s/configure/advanced", util.GetPackageHref(d.pkg, d.manifest)),
 		}, err))
-		checkTmplError(err, fmt.Sprintf("advanced-config (%s)", pkgName))
+		checkTmplError(err, fmt.Sprintf("advanced-config (%s)", d.manifestName))
 	} else if r.Method == http.MethodPost {
-		pkg.Spec.PackageInfo.Version = selectedVersion
-		if repositoryName != "" {
-			pkg.Spec.PackageInfo.RepositoryName = repositoryName
+		d.pkg.GetSpec().PackageInfo.Version = d.selectedVersion
+		if d.repositoryName != "" {
+			d.pkg.GetSpec().PackageInfo.RepositoryName = d.repositoryName
 		}
-		if err := s.pkgClient.ClusterPackages().Update(ctx, pkg); err != nil {
-			s.respondAlertAndLog(w, err,
-				fmt.Sprintf("An error occurred updating package %v to version %v in repo %v", pkgName, selectedVersion, repositoryName),
-				"danger")
-			return
+		if d.manifest.Scope.IsCluster() {
```

Comment:
I think it would be better to use a type switch here:
```go
switch pkg := d.pkg.(type) {
  case *v1alpha1.Package:
    // ...
  case *v1alpha1.ClusterPackage:
    // ...
  default:
    panic("unexpected package type")
}
```